src/main/java/br/com/monkey/sdk/jpa/MonkeyJpaSpecificationSupport.java

```diff
@@ -0,0 +1,11 @@
+package br.com.monkey.sdk.jpa;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+interface MonkeyJpaSpecificationSupport<T> {
+
+	Page<T> findAllUsingGroupBy(Specification<T> spec, Pageable pageable);
```

Comment:
cara, pensa melhor nisso, por favor findAllUsingGroupBy é genérico, isso aqui nao faz referencia a groupBy, pensa melhor em uma solução mas eficaz, mais clara e melhor pra daqui 1 ano quando alguém for usar esse cara, nao pensar coisas do tipo blz, UsingGroupBy, onde? cadê o parâmetro do groupBy, como eu passo o groupBy? será que nao tem uma maneira de no próprio findAll vc mudar o comportamento se existir groupBy na specification?